xbmc/video/videosync/VideoSyncCocoa.cpp

```diff
@@ -133,6 +133,7 @@ void CVideoSyncCocoa::UpdateFPS(double fps)
   {
     CLog::Log(LOGDEBUG, "CVideoSyncCocoa: Detected refreshrate: %i hertz", fpsInt);
     m_fps = fpsInt;
+    g_VideoReferenceClock.UpdateRefreshrate();
```

Comment:
this method is not supposed to be called from here. notify the while loop in Run() to exit if fps has changed.